pkg/grpc/api.go

```diff
@@ -1509,6 +1517,130 @@ func (a *api) SetMetadata(ctx context.Context, in *runtimev1pb.SetMetadataReques
+		apiServerLogger.Debug(err)
+		return &runtimev1pb.GetWorkflowResponse{}, err
+	}
+
+	id := &runtimev1pb.WorkflowReference{
+		InstanceID: response.WFInfo.InstanceID,
+	}
+
+	t, err := time.Parse(time.RFC3339, response.StartTime)
+	if err != nil {
+		panic(err)
+	}
+
+	pb := timestamppb.New(t)
+
+	res := &runtimev1pb.GetWorkflowResponse{
+		WorkflowInfo: id,
+		StartTime:    pb,
+		TaskQueue:    response.TaskQueue,
+		Status:       response.Status,
+	}
+	return res, nil
+}
+
+func (a *api) StartWorkflowAlpha1(ctx context.Context, in *runtimev1pb.StartWorkflowRequest) (*runtimev1pb.WorkflowReference, error) {
+	if in.WorkflowName == "" {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrWorkflowNameMissing))
+		apiServerLogger.Debug(err)
+		return &runtimev1pb.WorkflowReference{}, err
+	}
+
+	if in.WorkflowComponent == "" || a.workflows[in.WorkflowComponent] == nil {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrNoOrMissingComponent))
+		apiServerLogger.Debug(err)
+		return &runtimev1pb.WorkflowReference{}, err
+	}
+
+	if in.WorkflowInfo.InstanceID == "" {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrMissingOrEmptyInstance))
+		apiServerLogger.Debug(err)
+		return &runtimev1pb.WorkflowReference{}, err
+	}
+
+	wf := workflows.WorkflowReference{
+		InstanceID: in.WorkflowInfo.InstanceID,
+	}
+	req := workflows.StartRequest{
+		WorkflowInfo: wf,
+		Options:      in.Options,
+		WorkflowName: in.WorkflowName,
+		Parameters:   in.Parameters,
+	}
+
+	resp, err := a.workflows[in.WorkflowComponent].Start(ctx, &req)
+	if err != nil {
+		err = status.Errorf(codes.Internal, fmt.Sprintf(messages.ErrStartWorkflow, err))
+		apiServerLogger.Debug(err)
+		return &runtimev1pb.WorkflowReference{}, err
+	}
+	ret := &runtimev1pb.WorkflowReference{
+		InstanceID: resp.InstanceID,
+	}
+	return ret, nil
+}
+
+func (a *api) TerminateWorkflowAlpha1(ctx context.Context, in *runtimev1pb.TerminateWorkflowRequest) (*emptypb.Empty, error) {
+	if in.WorkflowInfo.InstanceID == "" {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrMissingOrEmptyInstance))
+		apiServerLogger.Debug(err)
+		return &emptypb.Empty{}, nil
+	}
+
+	if in.WorkflowComponent == "" {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrNoOrMissingComponent))
+		apiServerLogger.Debug(err)
+		return &emptypb.Empty{}, nil
+	}
+
+	if _, ok := a.workflows[in.WorkflowComponent]; !ok {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrComponentDoesNotExist, in.WorkflowComponent))
+		apiServerLogger.Debug(err)
+		return &emptypb.Empty{}, nil
+	}
+
+	req := workflows.WorkflowReference{
+		InstanceID: in.WorkflowInfo.InstanceID,
+	}
+
+	err := a.workflows[in.WorkflowComponent].Terminate(ctx, &req)
+	if err != nil {
+		err := status.Errorf(codes.InvalidArgument, fmt.Sprintf(messages.ErrTerminateWorkflow, err))
+		apiServerLogger.Debug(err)
+		return &emptypb.Empty{}, nil
```

Comment:
I noticed we're returning `nil` instead of `err` in all the error cases for `TerminateWorkflowAlpha1`. Is this intentional? I'm guessing not since `StartWorkflowAlpha1` and `GetWorkflowAlpha1` returns the error values.
```suggestion
		return &emptypb.Empty{}, err
```